app_dart/lib/src/request_handlers/check_flaky_tests_and_update_github.dart

```diff
@@ -0,0 +1,450 @@
+    final PullRequestBuilder prBuilder = PullRequestBuilder(stats: stats, issue: issue);
+    final GitCommit commit = await client.git.createCommit(
+      config.flutterflakybotSlug,
+      CreateGitCommit(
+        prBuilder.pullRequestTitle,
+        tree.sha,
+        parents: <String>[masterRef.object.sha],
+        author: commitUser,
+        committer: commitUser,
+      ),
+    );
+    final GitReference changeRef = await client.git.createReference(config.flutterflakybotSlug, ref, commit.sha);
+    final PullRequest pr = await client.pullRequests.create(
+      config.flutterSlug,
+      CreatePullRequest(
+        prBuilder.pullRequestTitle,
+        '${config.flutterflakybotSlug.owner}:${ref.replaceFirst(_refsPrefix, '')}',
+        'master',
+        body: prBuilder.pullRequestBody,
+      ),
+    );
+    await _assignReviewerFor(pr, reviewer: stats.testOwner, client: client);
+  }
+
+  Future<void> _assignReviewerFor(PullRequest pr, {String reviewer, GitHub client}) async {
+    const JsonEncoder encoder = JsonEncoder();
+    await client.postJSON<Map<String, dynamic>, PullRequest>(
+      '/repos/${config.flutterSlug.fullName}/pulls/${pr.number}/requested_reviewers',
+      convert: (Map<String, dynamic> i) => PullRequest.fromJson(i),
+      body: encoder.convert(<String, dynamic>{
+        'reviewers': <String>[reviewer],
+      }),
+    );
+  }
+
+  bool _isOtherIssueMoreImportant(Issue original, Issue other) {
+    // Open issues are always more important than closed issues. If both issue
+    // are closed, the one that is most recent created is more important.
+    if (original.isOpen && other.isOpen) {
+      throw 'There should not be two open issues for the same test';
+    } else if (original.isOpen && other.isClosed) {
+      return false;
+    } else if (original.isClosed && other.isOpen) {
+      return true;
+    } else {
+      return other.createdAt.isAfter(original.createdAt);
+    }
+  }
+
+  Future<String> _getCIContent(GitHub client) async {
+    final RepositoryContents contents = await client.repositories.getContents(config.flutterSlug, _ciYamlPath);
+    if (!contents.isFile) {
+      throw 'The path $_ciYamlPath should point to a file, but it is not!';
+    }
+    final String content = utf8.decode(base64.decode(contents.file.content.replaceAll('\n', '')));
+    return content;
+  }
+
+  Future<String> _getTestOwnerContent(GitHub client) async {
+    final RepositoryContents contents = await client.repositories.getContents(config.flutterSlug, _testOwnerPath);
+    if (!contents.isFile) {
+      throw 'The path $_ciYamlPath should point to a file, but it is not!';
+    }
+    final String content = utf8.decode(base64.decode(contents.file.content.replaceAll('\n', '')));
+    return content;
+  }
+
+  Future<String> _findTestOwner(String testName, String testOwnersContent) async {
+    final List<String> lines = testOwnersContent.split('\n');
```

Comment:
all `\n` has been replaced with ` `?

Reply:
the line 394 is to deal with the github format. the actual line break in `contents.file.content` is encoded in base64, so it won't be replace there